core/src/main/java/nl/nn/adapterframework/doc/model/ElementRole.java

```diff
@@ -16,22 +16,77 @@
 
 package nl.nn.adapterframework.doc.model;
 
+import java.util.Collection;
+import java.util.Comparator;
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
+import org.apache.logging.log4j.Logger;
+
+import lombok.AccessLevel;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import lombok.Setter;
 import nl.nn.adapterframework.doc.Utils;
+import nl.nn.adapterframework.util.LogUtil;
+
+public class ElementRole implements Comparable<ElementRole> {
+	private static Logger log = LogUtil.getLogger(ElementRole.class);
 
-public class ElementRole {
-	private final @Getter ElementType elementType;
+	private @Getter ElementType elementType;
 	private final @Getter String syntax1Name;
 	private final int syntax1NameSeq;
+	
+	// This property is used by FrankElement to calculate the ElementRole.isSuperseded
+	// property.
+	//
+	// It is calculated from the ConfigChild.isDeprecated property. This property is
+	// ambiguous by default, because the same ElementRole is reused by multiple
+	// ConfigChild objects. If multiple FrankElement have the same syntax 1 name and
+	// ElementType combination to determine children, then each time there is a different
+	// ConfigChild object but the ElementRole object is the same.
+	//
+	// The ambiguity is resolved by taking the value false when some
+	// related config children are deprecated while others aren't.
+	private @Getter boolean deprecated;
+
+	// This property is needed to avoid duplicate config children.
+	// As an example, consider Java class AbstractRecordHandler. It 
+	// has methods registerInputFields and registerChild, both taking
+	// an InputfieldsPart argument. By default, two of the config children of
+	// AbstractRecordHandler have the ElementRole-s: 
+	// * (elementType=InputfieldsPart, syntax1Name=inputFields) and
+	// * (elementType=InputfieldsPart, syntax1Name=child).
+	//
+	// Using both of these ElementRole to add config children would produce a
+	// conflict. Both would allow the same XML element, the XML tag
+	// that references Java class InputfieldsPart which is <InputFields>.
+	// When this conflict appears, one of the involved ElementRole is expected
+	// to be deprecated. That ElementRole then gets its superseded flag set.
```

Comment:
Het gaat niet om 'duplicates' of een conflict, maar om config children die onder verschillende rollen voor kunnen komen:
In Receiver heb je zo 'sender' en 'errorSender', en bestaan er dus de syntax-2 sub-elementen 'JavaSender' en 'JavaErrorSender'
Met deze 'superseded' functionaliteit heb je nu onder AbstractRecordHandler alleen het syntax-2 sub-element 'InputFields' en niet 'InputFieldsChild'. 

Dat lijkt sympathiek, maar het lijkt me toch niet juist om deze functionaliteit op te nemen. We hebben namelijk niet echt last van die extra elementen in de compatibility XSD, en als we zouden besluiten om errorSender deprecated te maken, dan zouden alle syntax2-style configuraties met errorSenders kapot gaan.